cpp/src/core/c_api.cpp

```diff
@@ -83,10 +86,14 @@ extern "C" cuvsError_t cuvsRMMFree(cuvsResources_t res, void* ptr, size_t bytes)
   });
 }
 
-thread_local std::unique_ptr<rmm::mr::pool_memory_resource<rmm::mr::cuda_memory_resource>> pool_mr;
+thread_local std::shared_ptr<
```

Comment:
I'm curious as to why this needs to be a shared_ptr instead of a unique_ptr like it was before?

Reply:
`rmm::mr::make_owning_wrapper` returns a `shared_ptr`. 
We could call the constructor directly to get a `unique_ptr` but not sure it's worth the extra complexity:
```
    using pool_wrapper =
      rmm::mr::owning_wrapper<rmm::mr::pool_memory_resource<rmm::mr::device_memory_resource>,
                              rmm::mr::device_memory_resource>;
    pool_mr = std::make_unique<pool_wrapper>(pool_wrapper(std::make_tuple(std::move(mr)),
                                                          std::forward<int>(initial_size),
                                                          std::forward<int>(max_size)));
```

Reply:
makes sense - thanks for the explanation!